functests/utils/pods/pods.go

```diff
@@ -87,6 +87,26 @@ func WaitForCondition(pod *corev1.Pod, conditionType corev1.PodConditionType, co
 	})
 }
 
+// WaitForPredicate waits until the given predicate against the pod returns true or error.
+func WaitForPredicate(pod *corev1.Pod, timeout time.Duration, pred func(pod *corev1.Pod) (bool, error)) error {
+	key := types.NamespacedName{
```

Comment:
now that we are know about `client.ObjectKeyFromObject` we can use it :)

Reply:
yep! will fix this